### PR TITLE
[mypyc] Basic test-only support for i32 and i64 

### PR DESCRIPTION
**Pull request by JukkaL**
Basic support for arithmetic, bitwise and comparison ops using native
int types `i32` and `i64`. Also support some implicit coercions between 
`int` and native int types. There are still various gaps in the functionality,
and that's why the feature is test-only so far.

Division and modulus ops require jumping through some hoops to make
the behavior with negative operands match Python semantics.

This can already show some improvements in benchmarks (using hacks
these can be used outside tests). I'll post figures once the implementation
is more complete.

Work on https://github.com/mypyc/mypyc/issues/837.

### CODE REVIEW

File: mypyc/test-data/irbuild-i64.test [hunk 1]
```diff
@@ -0,0 +1,1370 @@
...
+    r1 = load_mem r0 :: native_int*
+    keep_alive a
+    r2 = r1 << 1
+    r3 = r2 >> 1
+    r4 = n < r3 :: signed
+    if r4 goto L1 else goto L2 :: bool
+L1:
+    return 1
+L2:
+    return 0
```
Comment (ilevkivskyi) [inline mypyc/test-data/irbuild-i64.test:1370]: Couple ideas for tests:
* Fixed widths types in optional types
* Unions with different int types (for whatever we can support). I am however not sure what is reasonable here,  should we support narrowing unions by comparing value to some max/min value?

**Review by ilevkivskyi — APPROVED** [review]
Looks great, just have a bunch of questions/ideas.

File: mypyc/irbuild/ll_builder.py [hunk 2]
```diff
@@ -90,6 +95,11 @@
 # From CPython
 PY_VECTORCALL_ARGUMENTS_OFFSET: Final = 1 << (PLATFORM_SIZE * 8 - 1)
 
+FIXED_WIDTH_INT_BINARY_OPS: Final = {
+    '+', '-', '*', '//', '%', '&', '|', '^', '<<', '>>',
```
[the remark this answers is not in the document]
Comment (JukkaL) [reply mypyc/irbuild/ll_builder.py:99]: Currently it's not supported -- an explicit conversion to `float` is required. We can consider adding it once we have unboxed floats.

File: mypyc/irbuild/ll_builder.py [hunk 3]
```diff
@@ -1374,8 +1611,97 @@ def matching_call_c(self,
...
+            if is_int64_rprimitive(type):
+                prim = int64_mod_op
+            elif is_int32_rprimitive(type):
+                prim = int32_mod_op
+            else:
+                assert False, type
+            return self.call_c(prim, [lhs, rhs], line)
+        return self.int_op(type, lhs, rhs, op, line)
+
+    def inline_fixed_width_divide(self, type: RType, lhs: Value, rhs: Value, line: int) -> Value:
```
[the remark this answers is not in the document]
Comment (JukkaL) [reply mypyc/irbuild/ll_builder.py:1662]: I was able to share a bit of logic, but most of the logic is actually slightly different so sharing doesn't seem useful.

File: mypyc/test-data/exceptions.test [hunk 4]
```diff
@@ -514,3 +514,62 @@ L13:
...
+L0:
+    return 0
+def g():
+    r0 :: int64
+    r1 :: bit
+    r2 :: object
+    r3 :: int64
+L0:
+    r0 = f()
+    r1 = r0 == -113
```
[the remark this answers is not in the document]
Comment (JukkaL) [reply mypyc/test-data/exceptions.test:537]: Yes (in run tests we try returning all integer values in a range that includes -113).

File: mypyc/test-data/irbuild-i64.test (hunk 1 above)
Comment (JukkaL) [reply mypyc/test-data/irbuild-i64.test:1370]: Added an optional type test case. Unions with different int types don't make much sense, I think, so I didn't add a test case (yet).

**Pull request merged**